Sources/Player/PlaybackEngine/Internal/Events/PlayerEventSender.swift

```diff
@@ -77,10 +80,9 @@ class PlayerEventSender {
 	}
 
 	func send(_ offlinePlay: OfflinePlay) {
-		SafeTask { [weak self] in
-			guard let self else {
-				return
-			}
+		asyncSchedulerFactory.create { @MainActor [weak self] in
```

Comment:
[question] do we need the `@MainActor` here (and others below)?

Reply:
This was a good catch @ceciliasaraiva! 👏